src/rules/html/ensureRequiredAttributes.js

```diff
@@ -0,0 +1,46 @@
+import { HTML_TAGS_WITH_REQUIRED_ATTRIBUTES } from 'const';
+import * as messages from 'messages';
+
+
+export var ensureRequiredAttributes = ($, filename=null) => {
+  return new Promise((resolve, reject) => {
+    var promises = [];
+    var validatorMessages = [];
+
+    for (let tag in HTML_TAGS_WITH_REQUIRED_ATTRIBUTES) {
+      promises.push(_ensureAttributesInTag($, tag,
```

Comment:
Come to think of it this can be simpler; it doesn't need to use Promises here. I'll clean it up a bit.